examples/xonotic/go.mod

```diff
@@ -0,0 +1,19 @@
+module agones.dev/agones/examples/xonotic
+
+go 1.18
+
+require agones.dev/agones v0.0.0-dev
+
+require (
+	github.com/golang/protobuf v1.5.2 // indirect
+	github.com/grpc-ecosystem/grpc-gateway v1.11.3 // indirect
+	github.com/pkg/errors v0.9.1 // indirect
+	golang.org/x/net v0.0.0-20211209124913-491a49abca63 // indirect
+	golang.org/x/sys v0.0.0-20210616094352-59db8d763f22 // indirect
+	golang.org/x/text v0.3.6 // indirect
+	google.golang.org/genproto v0.0.0-20201110150050-8816d57aaa9a // indirect
+	google.golang.org/grpc v1.27.1 // indirect
+	google.golang.org/protobuf v1.26.0 // indirect
+)
+
+replace agones.dev/agones v0.0.0-dev => ../../
```

Comment:
@markmandel - what do you think about building this from head vs. a specific agones release? STK is currently built against 1.8.0, which is pretty old but also makes it clear the last time we updated it as well as making the build reproducible.

Reply:
🤔 hmnn, that's a good question.

The benefit I see of running against the development branch is that you can build out an example against something that is currently being worked on and not released.

That being said - pointing to a specific version ([like we do in STK](https://github.com/googleforgames/agones/blob/main/examples/supertuxkart/go.mod#L6)) is probably a better experience for the end user, so we should probably be consistent and do the same here.

It would also make the Docker build simpler as we don't have to force building from project root and copying in files (basically copy what we do in STK).

If we need to have in-development features, we can worry about it when it happens.

The only conflict I have - in simple-game-server we do probably want to continue building against a development version, as editing that binary is often necessary to implementing and testing new features. So that's likely the exception to the above rule.

Sound good?

Reply:
>pointing to a specific version ([like we do in STK](https://github.com/googleforgames/agones/blob/main/examples/supertuxkart/go.mod#L6)) is probably a better experience

I also feel that xonotic is pretty much similar STK example.

>If we need to have in-development features, we can worry about it when it happens.

Should we target to agones v1.8.0 (Works fine, just tested) or any other specific version?

Reply:
Let's use what is latest - `1.24.0`. Then we know when this example was built.